arangod/Aql/AsyncExecutor.h

```diff
@@ -42,7 +42,18 @@ class SharedQueryState;
 
 // The RemoteBlock is actually implemented by specializing ExecutionBlockImpl,
 // so this class only exists to identify the specialization.
-class AsyncExecutor final {};
+class AsyncExecutor final {
```

Comment:
All this structure looks unused except monostate, but it already was. Did I miss something?

Reply:
I just don't see their usage in the new diff

Reply:
Maybe I just missed something

Reply:
Yes, it's far from obvious. The `AsyncExecutor` was not used in the C++ tests until now. It now is, and uses the existing `ExecutorTestHelper` (e.g.
```cpp
  auto testHelper = makeExecutorTestHelper();
  testHelper
      .addDependency<AsyncExecutor>(registerInfos, {}, ExecutionNode::ASYNC)
      ...
```
). This needs the `Infos`, for example in `addDependency`:
```cpp
  template<typename E>
  auto addDependency(
      RegisterInfos registerInfos, typename E::Infos executorInfos,
      ExecutionNode::NodeType nodeType = ExecutionNode::SINGLETON)
      -> ExecutorTestHelper&;
```

And as I said, the Properties aren't currently necessary, but I think they're useful documentation of a block's behavior.